quarkchain/cluster/shard_db_operator.py

```diff
@@ -133,11 +139,22 @@ def get_transactions_by_address(
                 ]  # type: CrossShardTransactionDeposit
                 if should_skip(tx):
                     limit -= 1
+                    evm_tx = tx.tx.to_evm_tx()
+                    from_full_shard_id = evm_tx.quark_chain_config.get_full_shard_id_by_full_shard_key(
+                        tx.from_address.full_shard_key
+                    )
+                    to_full_shard_id = evm_tx.quark_chain_config.get_full_shard_id_by_full_shard_key(
+                        tx.to_address.full_shard_key
+                    )
+                    from_address = Address(
+                        tx.from_address.recipient, from_full_shard_id
+                    )
+                    to_address = Address(tx.to_address.recipient, to_full_shard_id)
                     tx_list.append(
                         TransactionDetail(
                             tx.tx_hash,
-                            tx.from_address,
-                            tx.to_address,
+                            from_address,
```

Comment:
a second thought, `TransactionDetail` should probably be as it is (without normalization) because that's the raw input. and the website should do its own logic to determine if it's `self` tx or not.